tests/test_permissionable_component.cairo

```diff
@@ -0,0 +1,200 @@
+    start_cheat_nonce, spy_events, EventSpyAssertionsTrait, ContractClassTrait, ContractClass,
+    start_cheat_block_timestamp, stop_cheat_block_timestamp
+};
+use core::hash::HashStateTrait;
+use core::pedersen::PedersenTrait;
+
+use token_bound_accounts::interfaces::IAccount::{
+    IAccountDispatcher, IAccountDispatcherTrait, IAccountSafeDispatcher, IAccountSafeDispatcherTrait
+};
+
+use token_bound_accounts::interfaces::IPermissionable::{
+    IPermissionableDispatcher, IPermissionableDispatcherTrait
+};
+
+use token_bound_accounts::interfaces::IExecutable::{
+    IExecutableDispatcher, IExecutableDispatcherTrait
+};
+use token_bound_accounts::interfaces::IUpgradeable::{
+    IUpgradeableDispatcher, IUpgradeableDispatcherTrait
+};
+use token_bound_accounts::components::presets::account_preset::AccountPreset;
+use token_bound_accounts::components::account::account::AccountComponent;
+
+use token_bound_accounts::components::permissionable::permissionable::PermissionableComponent;
+
+use token_bound_accounts::test_helper::{
+    hello_starknet::{IHelloStarknetDispatcher, IHelloStarknetDispatcherTrait, HelloStarknet},
+    erc721_helper::{IERC721Dispatcher, IERC721DispatcherTrait, ERC721},
+    simple_account::{ISimpleAccountDispatcher, ISimpleAccountDispatcherTrait, SimpleAccount}
+};
+
+const ACCOUNT1: felt252 = 5729;
+const ACCOUNT2: felt252 = 1234;
+const ACCOUNT3: felt252 = 6908;
+const ACCOUNT4: felt252 = 4697;
+
+// *************************************************************************
+//                              SETUP
+// *************************************************************************
+fn __setup__() -> (ContractAddress, ContractAddress) {
+    // deploy erc721 helper contract
+    let erc721_contract = declare("ERC721").unwrap();
+    let mut erc721_constructor_calldata = array!['tokenbound', 'TBA'];
+    let (erc721_contract_address, _) = erc721_contract
+        .deploy(@erc721_constructor_calldata)
+        .unwrap();
+
+    // deploy recipient contract
+    let account_contract = declare("SimpleAccount").unwrap();
+    let (recipient, _) = account_contract
+        .deploy(
+            @array![883045738439352841478194533192765345509759306772397516907181243450667673002]
+        )
+        .unwrap();
+
+    // mint a new token
+    let dispatcher = IERC721Dispatcher { contract_address: erc721_contract_address };
+    dispatcher.mint(recipient, 1.try_into().unwrap());
+
+    // deploy account contract
+    let account_contract = declare("AccountPreset").unwrap();
+    let mut acct_constructor_calldata = array![erc721_contract_address.try_into().unwrap(), 1, 0];
+    let (account_contract_address, _) = account_contract
+        .deploy(@acct_constructor_calldata)
+        .unwrap();
+
+    (account_contract_address, erc721_contract_address)
+}
+
+#[test]
+#[should_panic(expected: ('Account: invalid length',))]
+fn test_when_permissioned_addresses_and_permissions_not_equal() {
```

Comment:
rename test to `test_should_fail_if_unequal_permissioned_addresses_and_permissions`

Reply:
this is still yet to be renamed